src/loading.jl

```diff
@@ -104,3 +106,65 @@ function loadmodel!(dst, src; filter = _ -> true, cache = Base.IdSet())
 
   return dst
 end
+
+"""
+    state(x)
+
+Return an object with the same nested structure as `x` according to `Functors.children`, 
+but made only of basic containers (e.g. named tuples, tuples, arrays, and dictionaries).
+
+Besides trainable and non-trainable arrays, the state will contain leaf nodes that are not arrays,
+such as numbers, symbols, strings, and nothing values. The leaf types that end up in the state
+could increase in the future.
+
+This method is particularly useful for saving and loading models, 
+since the state contain only simple data types that can be easily serialized.
+
+The state can be passed to [`loadmodel!`](@ref) to restore the model.
+
+# Examples
+
+## Copy the state into another model
+
+```julia-repl
+julia> s = Flux.state(Dense(1, 2, tanh))
+(weight = Float32[0.5058468; 1.2398405;;], bias = Float32[0.0, 0.0], σ = missing)
+
+julia> m1 = Chain(Dense(1, 2, tanh), Dense(2, 1));
+
+julia> m2 = Chain(Dense(1, 2, tanh), Dense(2, 1));
+
+julia> s = Flux.state(m1)
+layers = ((weight = Float32[-0.56867087; 1.229064;;], bias = Float32[0.0, 0.0], σ = nothing), (weight = Float32[0.23323897 -0.5561147], bias = Float32[0.0], σ = nothing)),)
+
+julia> Flux.loadmodel!(m2, s);
+
+julia> m2[1].weight == m1[1].weight
+true
```

Comment:
IMO it would be worth showing some more complicated layers. Perhaps separately after the simple one is best. 
```suggestion
julia> m2[1].weight
2×1 Matrix{Float32}:
 1.0
 1.0

julia> Flux.state(trainmode!(Dropout(0.2)))  # stores p & activity, but not RNG state
(p = 0.2, dims = missing, active = true, rng = missing)

julia> Flux.state(BatchNorm(1))  # stores non-trainable arrays μ, σ²
(λ = missing, β = Float32[0.0], γ = Float32[1.0], μ = Float32[0.0], σ² = Float32[1.0], ϵ = 1.0f-5, momentum = 0.1f0, affine = true, track_stats = true, active = nothing, chs = 1)
```